src/Tizen.NUI.WindowSystem/src/public/SoftkeyService.cs

```diff
@@ -0,0 +1,312 @@
+                }
+            }
+        }
+
+        private Interop.SoftkeyService.SoftkeyExpandEventCallback _onExpandChanged;
+        private event EventHandler<ExpandState> _expandChanged;
+        /// <summary>
+        /// Emits the event when the expand state of the softkey service window is changed.
+        /// </summary>
+        /// <exception cref="ArgumentException">Thrown when failed of invalid argument.</exception>
+        /// <exception cref="Tizen.Applications.Exceptions.OutOfMemoryException">Thrown when the memory is not enough to allocate.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public event EventHandler<ExpandState> ExpandChanged
+        {
+            add
+            {
+                if (_expandChanged == null)
+                {
+                    _onExpandChanged = OnExpandChanged;
+                    int res = Interop.SoftkeyService.SetExpandEventHandler(_softkeyService, _onExpandChanged, IntPtr.Zero);
+                    _tzsh.ErrorCodeThrow(res);
+                }
+                _expandChanged += value;
+            }
+            remove
+            {
+                _expandChanged -= value;
+                if (_expandChanged == null)
+                {
+                    int res = Interop.SoftkeyService.SetExpandEventHandler(_softkeyService, null, IntPtr.Zero);
+                    _tzsh.ErrorCodeThrow(res);
+                }
+            }
+        }
+
+        private Interop.SoftkeyService.SoftkeyOpacityEventCallback _onOpacityChanged;
+        private event EventHandler<OpacityState> _opacityChanged;
+        /// <summary>
+        /// Emits the event when the opacity state of the softkey service window is changed.
+        /// </summary>
+        /// <exception cref="ArgumentException">Thrown when failed of invalid argument.</exception>
+        /// <exception cref="Tizen.Applications.Exceptions.OutOfMemoryException">Thrown when the memory is not enough to allocate.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public event EventHandler<OpacityState> OpacityChanged
+        {
+            add
+            {
+                if (_opacityChanged == null)
+                {
+                    _onOpacityChanged = OnOpacityChanged;
+                    int res = Interop.SoftkeyService.SetOpacityEventHandler(_softkeyService, _onOpacityChanged, IntPtr.Zero);
+                    _tzsh.ErrorCodeThrow(res);
+                }
+                _opacityChanged += value;
+            }
+            remove
+            {
+                _opacityChanged -= value;
+                if (_opacityChanged == null)
+                {
+                    int res = Interop.SoftkeyService.SetOpacityEventHandler(_softkeyService, null, IntPtr.Zero);
+                    _tzsh.ErrorCodeThrow(res);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Visible state of softkey.
+        /// </summary>
+        public enum VisibleState
```

Comment:
SoftkeyClient has same Enum types. How about sharing thease types.

Reply:
Unfortunately these enums are look same, but value is little different.
So we need to distinguish these enums.

Reply:
But it will confuse the user. I strongly recommend that extract this enum type to outside and share it. 
Each consumer (SoftkeyService, SoftkeyClient) can convert to internal enum type before calling native methods.

Reply:
I make ShellEnums.cs